### PR TITLE
Add job for checking for new Quilt dependencies

### PR DESCRIPTION
**Pull request by kklin**

### CODE REVIEW

File: config/jenkins/go.xml [hunk 1]
```diff
@@ -8,13 +8,13 @@ and copying the resulting file at
 <org.jenkinsci.plugins.golang.GolangBuildWrapper_-DescriptorImpl plugin="golang@1.2">
   <installations>
     <org.jenkinsci.plugins.golang.GolangInstallation>
-      <name>go-1.7</name>
+      <name>go-1.8</name>
```
Comment (kayousterhout) [inline config/jenkins/go.xml]: Why do the tests fail on version 1.7 / is this something our developer docs should point out?

File: config/jenkins/check-dependencies.xml [hunk 2]
```diff
@@ -0,0 +1,84 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<project>
+  <actions/>
+  <description>Check for any updates to our dependencies.</description>
```
Comment (kayousterhout) [inline config/jenkins/check-dependencies.xml]: What exactly will the failure message for this display? (Will it be obvious to the user that they just need to do "govendor fetch -v"? -- and if not is it easy to add that to the error msg?)

File: config/jenkins/check-dependencies.xml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+srcPath=$GOPATH/src/github.com/quilt/quilt
+mkdir -p ${srcPath}
+git clone https://github.com/quilt/quilt.git ${srcPath}
+
+cd ${srcPath}
+make get-build-tools
+
+# Fetch newest version of all vendor&apos;d packages, and see if resulting git tree is dirty.
+govendor fetch -v +vendor
```
Comment (kayousterhout) [inline config/jenkins/check-dependencies.xml]: should this also do a similar thing with "govendor remove +unused"?

File: config/jenkins/go.xml (hunk 1 above)
Comment (kklin) [inline config/jenkins/go.xml]: There's no good reason -- the tests do a direct string comparison of errors and the error message is different in go-1.8. We could be smarter in the unit tests.

Once we get a more formal release process we should think about what Go versions we'll support. I know that we'll be introducing a dependency on the sync map implementation in an upcoming Go release (https://trello.com/c/jt3ueAwG/728-on-the-next-release-version-of-go-golangorg-x-sync-syncmap-should-be-replaced-with-the-stdlib-version), though, so I'm not worried about not fully support 1.7 right now.

File: config/jenkins/check-dependencies.xml (hunk 2 above)
Comment (kklin) [inline config/jenkins/check-dependencies.xml]: There's no failure message exactly. Jenkins will just mark the build as failed, and then it's up to the person performing triage to look at the console output and figure out exactly why the build failed.

We could add smarter processing of errors, but I don't think it's worth the effort right now. I basically just want a job that makes sure dependency changes stay on my personal radar -- I'm not imagining that anyone could look at the job output and make the necessary changes.

File: config/jenkins/check-dependencies.xml (hunk 3 above)
Comment (kklin) [inline config/jenkins/check-dependencies.xml]: That sounds super useful, I'll add that!

**Comment by kklin** [comment]
Ready for another review @kayousterhout.

I made a number of changes:
- Check for `govendor remove +unused`
- Show what files changed using `git status`
- Run the job every 3 days rather than every day

File: config/jenkins/check-dependencies.xml (hunk 2 above)
Comment (kayousterhout) [inline config/jenkins/check-dependencies.xml]: To make the error slightly more digestible for a non-Kevin person, what about moving line 35 (the git status) to below the build, and right before that, adding something like `echo "ERROR: Vendoring is out of date (it is missing new updates and/or includes packages that are no longer necessary).  The correct vendoring includes these changes: "`

Comment (kklin) [inline config/jenkins/check-dependencies.xml]: Done!

**Pull request merged**